### PR TITLE
Initial prototype of artist similiarity

### PR DESCRIPTION
**Pull request by amCap1712**
Similar to recording similarity, add artist similarity based on listen sessions.

### CODE REVIEW

File: listenbrainz/labs_api/labs/api/similar_recordings.py [hunk 1]
```diff
@@ -27,28 +27,31 @@ def outputs(self):
...
-
         return {
             "type": "dataset",
             "columns": list(metadata[0].keys()),
             "data": metadata
         }
 
     def fetch(self, params, offset=-1, count=-1):
-        recording_mbids = [p["recording_mbid"] for p in params]
+        recording_mbids = [p["recording_mbid"].strip() for p in params]
```
Comment (mayhem) [inline listenbrainz/labs_api/labs/api/similar_recordings.py:54]: Yes, we need this here, but I think we need to add it to dataset hoster. :)

File: listenbrainz_spark/similarity/artist.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+def build_sessioned_index(listen_table, metadata_table, session, max_contribution, threshold, limit, _filter, skip_threshold):
+    # TODO: Handle case of unmatched recordings breaking sessions!
+    #  Detect and remove skips!
+    filter_artist_credit = "AND s1.artist_credit_id != s2.artist_credit_id" if _filter else ""
+    return f"""
+            WITH listens AS (
+                 SELECT user_id
+                      , BIGINT(listened_at)
+                      , CAST(COALESCE(recording_data.length / 1000, 180) AS BIGINT) AS duration
```
Comment (mayhem) [inline listenbrainz_spark/similarity/artist.py:22]: constants

**Review by mayhem — APPROVED** [review]
I'm glad you did the refactoring needed to make keep the code concise. Looking good!

File: listenbrainz/labs_api/labs/api/similar_recordings.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz/labs_api/labs/api/similar_recordings.py:54]: Makes sense.

File: listenbrainz_spark/similarity/artist.py (hunk 2 above)
Comment (amCap1712) [reply listenbrainz_spark/similarity/artist.py]: :+1:

**Pull request merged**